projects/js-packages/number-formatters/package.json

```diff
@@ -27,16 +27,23 @@
 		"jest-environment-jsdom": "29.7.0",
 		"typescript": "5.8.2"
 	},
+	"type": "module",
+	"main": "./dist/cjs/index.js",
+	"module": "./dist/mjs/index.js",
+	"types": "./dist/types/index.d.ts",
 	"exports": {
 		".": {
 			"jetpack:src": "./src/index.ts",
-			"types": "./dist/index.d.ts",
-			"default": "./dist/index.js"
+			"types": "./dist/types/index.d.ts",
+			"import": "./dist/mjs/index.js",
+			"require": "./dist/cjs/index.js"
 		}
 	},
 	"scripts": {
-		"build": "pnpm run clean && pnpm run compile-ts",
-		"clean": "rm -rf dist/",
+		"build": "pnpm run clean && pnpm run build:esm && pnpm run build:cjs",
+		"build:esm": "tsc --declarationDir dist/types --outDir dist/mjs",
+		"build:cjs": "tsc --module CommonJS --moduleResolution node10 --declaration false --outDir dist/cjs",
```

Comment:
I'm pretty wary of this too. https://www.typescriptlang.org/docs/handbook/modules/reference.html#commonjs, for example, says "You probably shouldn’t use this."

It seems the more modern way to do something like this, in the absence of offical support in tsc, is to somehow munge the "type" in package.json just before the build to have tsc with "nodenext" build the cjs version, then restore it after. For example, it looks like [@knighted/duel](https://www.npmjs.com/package/@knighted/duel) does that directly, while [tshy](https://www.npmjs.com/package/tshy) assumes sources are in `src/` and creating a temporary stub `src/package.json` will manage to DTRT. (of those two, BTW, duel seems more of the "do one specific thing" philosophy while tshy seems more "try to do all the things".)

Reply:
I think I addressed both concerns by creating a special `src/package.json` during the CommonJS build, with `type: 'commonjs'`. Just like `tshy` does. It's easy to do, just a `cp` and `rm` command.

We don't need to remove `customConditions` any more, because `module` and `moduleResolution` are always `nodenext`. It's the `type` field that triggers the CommonJS build.